src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -122,10 +124,18 @@ var ReactCompositeComponentMixin = {
    * @final
    * @internal
    */
-  mountComponent: function(rootID, transaction, context) {
+  mountComponent: function(
+    rootID,
+    transaction,
+    nativeParent,
+    nativeContainerInfo,
+    context
+  ) {
     this._context = context;
     this._mountOrder = nextMountID++;
     this._rootNodeID = rootID;
+    this._nativeParent = nativeParent;
+    this._nativeContainerInfo = nativeContainerInfo;
```

Comment:
I realized this isn't necessary only at the root. Your way doesn't work if we don't store this here because _nativeParent is also a ReactDOMComponent. We could crawl up the tree every time but that feels more wasteful. Essentially, this is where we store the info that doesn't change within the tree but that we need at every level.